File: Strings/palindrome.py
def is_palindrome(word):
    #menghapus sepasi dan mengubah huruf menjadi kecil
    clear_word =  word.lower().replace(" ", "")
    return clear_word == clear_word[::-1]

while True:
    kata_input = str(input("Masukan kata (Q for exit): "))
    if kata_input == "q": 
        print("You exited the program")
    if is_palindrome(kata_input):
        print(f"{kata_input} adalah palindrome")
    else:
        print(f"Bukan, {kata_input} adalah palindrome")

File: Strings/palindrome.cpp
#include <iostream>
#include <cstring>
using namespace std;

//fungsi global
bool isPalindrome(const string &word)
{
    int length = word.length();
    int i, j;
    for(i = 0, j = length - 1; i < j; i++, j--)
    {
        if(word[i] != word[j]){
            return false;
        }
    }

    return true;
}

int main(){
    string kata;

    while (true)
    {
        cout <<"Masukan kata (Q for exit): ";
        cin >> kata;

        if (kata == "q")
        {
            cout <<"You exited the program" << endl;
            break;
        }
        

        if (isPalindrome(kata))
        {
            cout << kata <<" adalah palindrome" << endl;
        }
        else
        {
            cout << kata <<" adalah bukan palindrome" << endl;
        }
    }

    return 0;
}